relations/src/operation.rs

```diff
@@ -0,0 +1,48 @@
+use halo2_base::{utils::BigPrimeField, AssignedValue};
+
+/// A trait representing an operation.
+pub trait Operation<F>
+where
+    Self: Sized,
+    F: BigPrimeField,
+{
+    /// The private data associated with the operation.
+    type OpPriv;
+    /// The public data associated with the operation.
+    type OpPub;
+
+    /// Combines the private and public data to create an instance of the operation.
+    ///
+    /// # Arguments
+    ///
+    /// * `op_priv` - The private data of the operation.
+    /// * `op_pub` - The public data of the operation.
+    ///
+    /// # Returns
+    ///
+    /// An `Option` containing the combined operation, or `None` if the combination fails.
+    fn combine(op_priv: Self::OpPriv, op_pub: Self::OpPub) -> Option<Self>;
+}
+
+pub trait CircuitOperation<F>
```

Comment:
this is literally the same trait as `Operation`, right? so maybe we don't need both

Reply:
I'm not sure. This one was created as a partial placeholder as there have to be flow for example `OpPub -> AssignedValue -> CircuitOpPub`, but more extended tests have to be written to check how this trait should exatly look.